### PR TITLE
feat(edge): add mTLS support

### PR DESCRIPTION
**Pull request by deviantony**
Replaces https://github.com/portainer/agent/pull/234

Rebase the PR above on top of the ce-2.13 branch and contain only changes required for mTLS support on the agent side.

Requires https://github.com/portainer/portainer/pull/6529

### CODE REVIEW

**Comment by SvenDowideit** [comment]
yup, that looks like the changes I made for this - nice extraction

LGTM

File: edge/stack/stack.go [hunk 1]
```diff
@@ -72,9 +72,8 @@ type StackManager struct {
 }
 
 // NewStackManager returns a pointer to a new instance of StackManager
-func NewStackManager(portainerURL, endpointID, edgeID string, insecurePoll bool) (*StackManager, error) {
-	cli := client.NewPortainerClient(portainerURL, endpointID, edgeID, insecurePoll)
-
+// TODO: REVIEW no point in returning error
```
Comment (andres-portainer) [inline edge/stack/stack.go]: I agree, we should remove the error.

File: edge/edge.go [hunk 2]
```diff
@@ -220,3 +243,52 @@ func (manager *Manager) checkDockerRuntimeConfig() error {
...
+			}
+			caCertPool = x509.NewCertPool()
+			caCertPool.AppendCertsFromPEM(caCert)
+		}
+
+		// Create a HTTPS client and supply the created CA pool and certificate
+		return &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs:      caCertPool,
+				Certificates: []tls.Certificate{cert},
```
Comment (andres-portainer) [inline edge/edge.go:262]: Can we use TLS 1.3 here please please?

```
MinVersion: tls.VersionTLS13,
MaxVersion: tls.VersionTLS13,
```

**Review by andres-portainer — APPROVED** [review]

**Review by cheloRydel — APPROVED** [review]

**Comment by cheloRydel** [comment]
Replaced by https://github.com/portainer/agent/pull/261

**Pull request closed without merging**